src/sgl/device/command.cpp

```
@@ -968,13 +981,33 @@ std::string CommandEncoder::to_string() const
     , m_rhi_command_buffer(std::move(command_buffer))
 {
 }
 
-CommandBuffer::~CommandBuffer() { }
+CommandBuffer::~CommandBuffer()
+{
+    if (!m_submitted)
+        m_device->_notify_command_recording_discarded(m_recording_id);
+}
+
+void CommandBuffer::_notify_submitted(uint64_t submit_id) noexcept
```

Cppcheck (2.20.0)

[style] The function '_notify_submitted' is never used.

(unusedFunction)